File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        """
        :type piles: List[int]
        :type h: int
        :rtype: int
        """
        # Binary search for the minimum value of k
        left, right = 1, max(piles)
        while left < right:
            mid = (left + right) // 2
            if sum((p - 1) // mid + 1 for p in piles) > h:
                left = mid + 1
            else:
                right = mid

        return left

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l=1,r=1e9,cur=-1,md;
        while(l<=r){
            md=(l+r)>>1;
            long long time = 0;
            for(auto t : piles)
                time+=(t+md-1)/md;
            if(time<=h){
                r=md-1;cur=md;
            }
            else l=md+1;
        }
        return cur;
    }
};